discretisedfield/tests/test_field.py

```diff
@@ -3160,14 +3160,19 @@ def test_mpl(test_field):
     plt.close("all")
 
 
-def test_mpl_dimension(valid_mesh):
-    field = df.Field(valid_mesh, nvdim=1)
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+def test_mpl_dimension(valid_mesh, nvdim):
+    field = df.Field(valid_mesh, nvdim=nvdim)
 
-    if valid_mesh.region.ndim != 2:
+    if valid_mesh.region.ndim != 2 or nvdim > 3:
         with pytest.raises(RuntimeError):
-            field.mpl.scalar()
+            field.mpl()
     else:
-        field.mpl.scalar()
+        if nvdim == 3:
+            field.vdim_mapping = dict(
+                zip(field.vdims, [*valid_mesh.region.dims, None, None])
+            )
```

Comment:
I think this is in principle good.

We should only need one additional `None` as `dims==2`.
```suggestion
            field.vdim_mapping = dict(
                zip(field.vdims, [*valid_mesh.region.dims, None])
            )
```